src/test/java/unit/uk/gov/ida/verifyserviceprovider/services/NonMatchingAssertionServiceTest.java

```diff
@@ -91,16 +91,17 @@ public void setUp() {
         IdaSamlBootstrap.bootstrap();
         initMocks(this);
 
-        assertionValidator = new AssertionValidator(instantValidator,subjectValidator,conditionsValidator);
-
         nonMatchingAssertionService = new NonMatchingAssertionService(
                 hubSignatureValidator,
                 subjectValidator,
-                attributeStatementValidator
+                attributeStatementValidator,
+                authnContextFactory,
+                verifyMatchingDatasetUnmarshaller,
+                assertionClassifier
         );
         doNothing().when(subjectValidator).validate(any(), any());
         when(hubSignatureValidator.validate(any(), any())).thenReturn(mock(ValidatedAssertions.class));
-
+        when(assertionClassifier.classifyAssertion(any())).thenCallRealMethod();
```

Comment:
Might as well just instantiate this as a real class. I don't think this hurts the tests too much.